test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -33,24 +33,27 @@
 from torch.testing._internal import common_distributed, common_utils
 from torch.testing._internal.common_utils import IS_WINDOWS
 
-if IS_WINDOWS:
-    print("Test fails on windows, see https://github.com/pytorch/pytorch/issues/63086")
-    sys.exit(0)
-
 try:
     import torchvision
     HAS_TORCHVISION = True
 except ImportError:
     HAS_TORCHVISION = False
 
 # Use GLOO on GPU when running CUDA + Windows
-BACKEND = (
-    dist.Backend.NCCL if not IS_WINDOWS and torch.cuda.is_available()
-    else dist.Backend.GLOO
-)
+def _get_backend_for_tests():
```

Comment:
curious, is this a better convention than the existing code?

Reply:
I mainly needed it for L686, it would look strange if we had `backend = BACKEND if use_gpu else GLOO`, so went with this instead.